### PR TITLE
arcane tampered door fix

### LINKED ISSUE #36905: Arcane Tamper
(WEB REPORT BY: adacovsk REMOTE: 172.93.101.77:7777)
# Revision
43c624c4e3020c678bfe77e0fdee1878a615c4a2

# Byond Version
515.1641

# Description
arcane tamper on a door and it does the steam but no "D'R ST'K!"

Line 397 according to Kavlax
# Steps to Reproduce

# What you Expected
go through door and be teleported
# What Actually Happened
just a normal door.

Maybe have it reroll the door after casting the spell on the door again, or remove certain doors where it causes the spell to fail

### PR DESCRIPTION
**Pull request by adacovsk**
- sometimes some arcane tampered doors don't work (the spell fails) and the door works as normal and you can't tamper them again

## What this does
- makes the code more readable
- might make the code slightly faster? (O(1) most likely, O(n) at worst, as opposed to copy lists and iterating through whole list)
- allows for doors to be tampered again

## Why it's good
- it's no fun if your spell fails and you can't cast the spell on the door again
- Closes #36905

## How it was tested
- tampered like 50 doors
image

## Changelog

:cl:
 * bugfix: can cast arcane tamper on doors again after already tampered/bugged out

### CODE REVIEW

File: code/game/machinery/doors/door.dm [hunk 1]
```diff
@@ -399,16 +399,42 @@ var/list/all_doors = list()
...
+			arcane_linked_door = candidate
+			candidate.arcanetampered = arcanetampered
+			candidate.arcane_linked_door = src
+			return "D'R ST'K!"
+		
+		// Door was invalid - add to exclusion list for next iteration
+		tested_invalid += candidate
+		attempts++
+	
+	return ""
```
Comment (west3436) [inline code/game/machinery/doors/door.dm:437]: I am not sure if returning an empty string will give us a reasonable result. Can you please test what happens when this is casted on something that would return an empty string?

**Review by west3436 — CHANGES_REQUESTED** [review]

**Comment by SonixApache** [comment]
i'm not fully familiar with the spell but also test if de-tampering and re-tampering works

**Comment by adacovsk** [comment]
> i'm not fully familiar with the spell but also test if de-tampering and re-tampering works

will test

File: code/game/machinery/doors/door.dm (hunk 1 above)
Comment (adacovsk) [reply code/game/machinery/doors/door.dm:437]: null string causes the wizard to say nothing when it's cast, and the door doesn't work (edge case). We still get the special effects around the door though...

**Comment by adacovsk** [comment]
> i'm not fully familiar with the spell but also test if de-tampering and re-tampering works

blessing and recasting on the door works

File: code/game/machinery/doors/door.dm (hunk 1 above)
Comment (adacovsk) [reply code/game/machinery/doors/door.dm:437]: I could probably remove `!is_blocked_turf(T, src)` from `arcane_linkable()` and it would fix all these issues, but I don't know the consequences

**Comment by PrimeDSS13** [comment]
Is this code AI?

**Comment by adacovsk** [comment]
> Is this code AI?

This is beautifully written code (partially, yes, with lots of direction)

**Comment by angryturnip** [comment]
I thought you said future PRs wouldn't use AI

**Comment by adacovsk** [comment]
> I thought you said future PRs wouldn't use AI

AI agents specifically

**Comment by falcon2346** [comment]
> AI agents specifically

I hope this means you've hired an indian prompt engineer to code for you

**Comment by angryturnip** [comment]
> AI agents specifically

That's not what you said

**Comment by west3436** [comment]
> > I thought you said future PRs wouldn't use AI
> 
> AI agents specifically

Don't be pedantic.

Acceptable LLM use:
- Using it to help you fix a proc you're working on.
- Using it to help debug your code.
- Providing examples or demo code for a tough problem you're working on.

Bad use:
- "@grok solve this runtime"
- Not actually knowing what the code is doing or even how to solve the issue yourself

**Comment by adacovsk** [comment]
Claude Code (Anthropic's AI agent) was giving me poor results, which is why I'm no longer using it. It didn't follow the proper formatting for PRs, didn't link relevant issues, and since it is constantly running in the background, it is more likely to hallucinate, especially with custom codebases. Wouldn't recommend grok for this; Claude Opus or Sonnet are more geared for coding.

**Comment by d3athrow** [comment]
I doubt you bothered to test any of your ai generated fixes

**Comment by SonixApache** [comment]
@d3athrow 
why'd you close this
when we know he tested it
